editor/editor_resource_picker.cpp

```diff
@@ -134,16 +140,25 @@ void EditorResourcePicker::_file_selected(const String &p_path) {
 	if (!base_type.is_empty()) {
 		bool any_type_matches = false;
 
-		for (int i = 0; i < base_type.get_slice_count(","); i++) {
-			String base = base_type.get_slice(",", i);
-			if (loaded_resource->is_class(base)) {
-				any_type_matches = true;
-				break;
+		String res_type = loaded_resource->get_class();
+		Ref<Script> res_script = loaded_resource->get_script();
+		bool is_global_class = false;
+		if (res_script.is_valid()) {
+			String script_type = EditorNode::get_editor_data().script_class_get_name(res_script->get_path());
+			if (!script_type.is_empty()) {
+				is_global_class = true;
+				res_type = script_type;
 			}
 		}
 
+		for (int i = 0; i < base_type.get_slice_count(",") && !any_type_matches; i++) {
+			String base = base_type.get_slice(",", i);
+
+			any_type_matches = is_global_class ? EditorNode::get_editor_data().script_class_is_parent(res_type, base) : loaded_resource->is_class(base);
```

Comment:
Should break on match hit.

Reply:
Do you mean to say that I should explicitly use a `break` inside of an `if` statement rather than making the `any_type_matches` variable part of the loop condition, as I have done?